### PR TITLE
LPS-90616, new query subsystem

### PR DESCRIPTION
**Pull request by rbohl**
https://issues.liferay.com/browse/LPS-90616, queries draft

https://issues.liferay.com/browse/LPS-90870

### CODE REVIEW

**Comment by lipusz** [comment]
**Started reviewing...** :mag:

**Comment by lipusz** [comment]
It's a minor thing, though wanted to ask if it's intended that you used LPS-90616 (the ticket for the story) and not LPS-90870 (the ticket for the documentation)?

File: developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown [hunk 1]
```diff
@@ -0,0 +1,186 @@
+---
+header-id: queries-and-filters
+---
+
+# Queries and Filters
```
Comment (lipusz) [inline developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown]: From search/discovery point of view, "Search Queries and Filters" would be better. I know that they are under the "search" structure in the folder hierarchy, but this information is not something appearing in the title and the full path may not be "indexed" for search so developers may need to dig more to get to this important page.

What do you think?

@rbohl

File: developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+As the ancient wisdom holds, 
+
+> A search without a query is like an answer without a question.
+
+To get sensible results from the search engine, you must provide a sensible
+query. 
+
+## The Liferay Search Application's Search Query
+
+To see how the @product@ Search application constructs its main query string,
```
Comment (lipusz) [inline developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown]: Here, we are mixing 2 things:
* Search widgets
* Search API

Search widgets are also built on top of our Search APIs, though they provide a certain feature using the queries and filters. We can show the final query constructed for Search bar searches using the Insights widgets and that's all.

The other part is Search API in general. I think the purpose of this particular developer guide is to show examples for API usage, but not to be too specific so maybe it's not best idea to start with this topic.

What we could do is to reference certain locations from the code where we use a given query or filter "in action", but showing our "big boolean query monster" may be not useful within this context.

File: developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+`com.liferay.portal.search.elasticsearch6.internal.query.ElasticsearchQueryTranslator`?
+
+## Queries and Filters, at the Search Engine Level
+
+Elasticsearch and Solr do not make API level distinctions between queries and
+filters. Since they're each used in different circumstances, Liferay's API
+explicitly provides two sets of APIs, one for queries and one for filters. Both
+APIs are in the `portal-search-api` module.
+
+*Filters* ask a yes or no question for every document.
```
Comment (lipusz) [inline developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown]: but do not contribute to the score (relevancy). It's like limiting the document set my query will run on, for example: filtering documents for a given "Virtual Instance" by the `companyId` field.

**Comment by rbohl** [comment]
> It's a minor thing, though wanted to ask if it's intended that you used LPS-90616 (the ticket for the story) and not LPS-90870 (the ticket for the documentation)?

No, that was unintentional.

File: developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown (hunk 1 above)
Comment (rbohl) [reply developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown]: Yes that's a good idea :nerd_face:

File: developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown (hunk 2 above)
Comment (rbohl) [reply developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown]: Okay, sounds good. I'll start poking around, but if you know a few good examples feel free to contribute them.

File: developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown (hunk 3 above)
Comment (rbohl) [reply developer/frameworks/articles/160-search/xx-queries-and-filters/01-queries-and-filters-intro.markdown]: Got it. I thought we made this clear in the draft, but I'll go make sure it's properly emphasized.

**Comment by lipusz** [comment]
Review edits here: https://github.com/lipusz/liferay-docs/tree/LPS-90616-search-queries-7.2

**Comment by lipusz** [comment]
https://github.com/rbohl/liferay-docs/pull/115

**Pull request closed without merging**